### PR TITLE
500 is sometimes the right answer

### PR DESCRIPTION
**Pull request by mccalluc**
@afred? I get failures locally because of AWS. Think I should also add a couple new tests.

### CODE REVIEW

**Comment by mccalluc** [comment]
Second thought: the tests I was thinking of are already in place:

```ruby
    it 'Gives 404 for bad query' do
      visit '/catalog?f[access]='
      expect(page.status_code).to eq(404)
    end
  end

  ...

    it 'Gives 404 for bad item' do
      visit '/catalog/nope'
      expect(page.status_code).to eq(404)
    end
```

File: app/controllers/tabbed_controller.rb [hunk 1]
```diff
@@ -8,9 +8,11 @@ def show
...
 
     if params[:tab]
-      fail ActiveRecord::RecordNotFound unless @item.tabs[params[:tab]]
+      fail IndexError unless @item.tabs[params[:tab]]
     else
       redirect_to('/' + params[:controller] + '/' + @item.tab_path) if @item.tab_path != params[:id]
     end
   end
+
+  rescue_from IndexError, with: :render_404
```
Comment (afred) [inline app/controllers/tabbed_controller.rb]: @mccalluc is `params[:tab]` a string, or an integer?

Comment (mccalluc) [inline app/controllers/tabbed_controller.rb]: string: It's coming from the routes. I have no strong preferences on what the particular error is... You could argue for making a new class that is only used here, so we can be absolutely sure what we're catching.

**Comment by afred** [comment]
@mccalluc see my modifications.

`IndexError` is too generic an exception to assume we want to throw a 404 whenever we find it. However, it's a nuance of CMLess that `IndexError` gets thrown when it can't find something. That's fine, but it's best to tighten up when we're looking for that.. which is what I did in `OverrideController`.

For `TabbedController`, we don't need to bring any exceptions into it at all really. We just want to render 404 if you're trying to get to a non-existent tab.

File: app/controllers/tabbed_controller.rb [hunk 2]
```diff
@@ -8,11 +8,9 @@ def show
...
 
     if params[:tab]
-      fail IndexError unless @item.tabs[params[:tab]]
+      render_404 unless @item.tabs[params[:tab]]
     else
       redirect_to('/' + params[:controller] + '/' + @item.tab_path) if @item.tab_path != params[:id]
     end
   end
-
-  rescue_from IndexError, with: :render_404
```
Comment (mccalluc) [inline app/controllers/tabbed_controller.rb]: @afred: Without this, we'll be missing IndexErrors thrown by CMLess... but as you note, CMLess should really be more precise. So this is fine, and when we have a more precise error, we'll add it.

**Pull request merged**